tests/tests/users/sign-up.spec.ts

```diff
@@ -101,6 +101,7 @@ test.describe('user invited to 2 courses', () => {
 
     await page.goto(`/courses/${course1.id}`);
     await expect(page).toHaveURL(new RegExp(course1.id));
+    await page.waitForTimeout(1000);
```

Comment:
Does this solve a separate problem? This does not seem related to soft-deletion of course_user.

Fixing flaky tests should be in a separate PR, and you aren't explaining here why we now need to have this `waitForTimeout`.  
https://playwright.dev/docs/api/class-page#page-wait-for-timeout quoting the docs:  
> Tests using the timer in production are going to be flaky. Use signals such as network events, selectors becoming visible and others instead.

Reply:
Ok, its not related. Lemme revert it.